support-lib/jni/Marshal.hpp

```diff
@@ -607,12 +612,22 @@ namespace djinni
         {
             // Serialize to C++ vector
             std::vector<uint8_t> cppBuf(c.ByteSizeLong());
-            [[maybe_unused]]
-            bool success = c.SerializeToArray(cppBuf.data(), static_cast<int>(cppBuf.size()));
-            assert(success);
+            LocalRef<jobject> javaBuf;
+
+            if (cppBuf.size() > 0) {
+                [[maybe_unused]]
+                bool success = c.SerializeToArray(cppBuf.data(), static_cast<int>(cppBuf.size()));
+                assert(success);
+
+                // Wrap C++ vector as java.nio.ByteBuffer (no ownership)
+                javaBuf = LocalRef<jobject>(jniEnv, jniEnv->NewDirectByteBuffer(cppBuf.data(), cppBuf.size()));
+            } else {
+                // Can't have a zero-length DirectByteBuffer so use an empty Java array-backed ByteBuffer
+                const auto& byteBuffer = JniClass<ByteBufferJniInfo>::get();
+                javaBuf = LocalRef<jobject>(jniEnv, jniEnv->CallStaticObjectMethod(byteBuffer.clazz.get(), byteBuffer.method_allocate, 0));
```

Comment:
All zero size buffers are the same, so maybe we can create once into a static variable, and reuse it.